# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerNavigator.h
# Commit: 244df402d6eb
# Full Hash: 244df402d6ebc15d6845c4144245511021408835
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1131324
# File Overlap Count: 1
# Description:
#   Bug 1131324 - Expose ServiceWorkerContainer on WorkerNavigator. r=dom-worker-reviewers,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213725
# ==============================================================================

diff -r 57f70ba5f27e -r 244df402d6eb dom/workers/WorkerNavigator.h
--- a/dom/workers/WorkerNavigator.h	Thu Oct 24 03:02:39 2024 +0000
+++ b/dom/workers/WorkerNavigator.h	Thu Oct 24 03:02:40 2024 +0000
@@ -32,6 +32,7 @@
 class MediaCapabilities;
 class LockManager;
 class Permissions;
+class ServiceWorkerContainer;
 
 namespace network {
 class Connection;
@@ -48,6 +49,7 @@
   RefPtr<webgpu::Instance> mWebGpu;
   RefPtr<dom::LockManager> mLocks;
   RefPtr<dom::Permissions> mPermissions;
+  RefPtr<ServiceWorkerContainer> mServiceWorkerContainer;
   bool mOnline;
 
   WorkerNavigator(const NavigatorProperties& aProperties, bool aOnline);
@@ -117,6 +119,8 @@
   dom::LockManager* Locks();
 
   dom::Permissions* Permissions();
+
+  already_AddRefed<ServiceWorkerContainer> ServiceWorker();
 };
 
 }  // namespace dom